### PR TITLE
Add progress logging for JVM toolchain discovery

### LINKED ISSUE #23428: Display a progress message when probing Java toolchains
### Expected Behavior

If you use Java toolchains, for example:

```gradle
java {
    toolchain {
        languageVersion = JavaLanguageVersion.of(11)
    }
}
```

Then Gradle will probe (and cache) the result of probing each of the recognized JVMs, but it won't indicate in the progress bar that it's doing so.

### Current Behavior

When you have a lot of JDKs installed in "standard" locations that Gradle can detect (e.g SDKMan!), probing the JVMs can take a few seconds, resulting in a build apparently "stuck" when it's actually probing the JVMs.

### Context

This scan shows a very basic app which doesn't use toolchains: https://scans.gradle.com/s/x335gxpgydqhg

Now add the following code snippet to your build script:

```gradle
java {
    toolchain {
        languageVersion = JavaLanguageVersion.of(11)
    }
}
```

In this [build scan](https://scans.gradle.com/s/ldafezebxmwzg) it only took 2s, but in my previous test it took up to 11s (I guess because there's a cache of probed JDKs living somewhere or that now I have my OS caches warm).

### PR DESCRIPTION
**Pull request by ov7a**
Fixes #23428

### CODE REVIEW

**Comment by ov7a** [comment]
@bot-gradle test

**Review by ljacomet — CHANGES_REQUESTED** [review]
This looks good overall, see below for a few minor comments and a possibly broken test.

I would however add a note in toolchains auto detection documentation about this information being printed. And then a short entry in release notes.

File: subprojects/jvm-services/src/main/java/org/gradle/jvm/toolchain/internal/LocationListInstallationSupplier.java
Comment (ov7a) [reply subprojects/jvm-services/src/main/java/org/gradle/jvm/toolchain/internal/LocationListInstallationSupplier.java:44]: Me too, but I took these strings "as is" from existing source names. They are displayed in toolchains report, and I wasn't sure whenever it is safe to edit them.

File: subprojects/platform-jvm/src/integTest/groovy/org/gradle/jvm/toolchain/ShowToolchainsIntegrationTest.groovy
Comment (ov7a) [reply subprojects/platform-jvm/src/integTest/groovy/org/gradle/jvm/toolchain/ShowToolchainsIntegrationTest.groovy:55]: It looks like that `registerJavaInstallationSuppliers` register all toolchain suppliers without any OS filtering, but you are right, this might be changed. I'll replace this to something else.

**Comment by ov7a** [comment]
Note that IDE integration isn't changed
image

**Review by ljacomet — APPROVED** [review]
LGTM!

You can merge once you resolve the conflicts without another review IMO

File: subprojects/code-quality/src/integTest/groovy/org/gradle/api/plugins/quality/codenarc/CodeNarcPluginToolchainsIntegrationTest.groovy
Comment (ljacomet) [inline subprojects/code-quality/src/integTest/groovy/org/gradle/api/plugins/quality/codenarc/CodeNarcPluginToolchainsIntegrationTest.groovy:36]: Given the conversations, this is not a satisfying solution. Either pull that change from this PR to not delay it further, or apply one of the suggestions.

**Review by ljacomet — CHANGES_REQUESTED** [review]

**Review by ljacomet — APPROVED** [review]
LGTM!

Can be merged once conflicts are resolved.

**Comment by ov7a** [comment]
@bot-gradle test and merge

**Pull request merged**
